embeddings_test.go

```diff
@@ -89,3 +89,41 @@ func TestEmbeddingEndpoint(t *testing.T) {
 	_, err = client.CreateEmbeddings(ctx, EmbeddingRequest{})
 	checks.NoError(t, err, "CreateEmbeddings error")
 }
+
+func TestCosineSimilarity(t *testing.T) {
+	v1 := []float32{1, 2, 3}
+	v2 := []float32{2, 4, 6}
+	expected := float32(1.0)
+
+	result := CosineSimilarity(v1, v2)
+	if result != expected {
+		t.Errorf("Unexpected result. Expected: %v, but got %v", expected, result)
+	}
+
+	v1 = []float32{1, 0, 0}
+	v2 = []float32{0, 1, 0}
+	expected = float32(0.0)
+	result = CosineSimilarity(v1, v2)
+
+	if result != expected {
+		t.Errorf("Unexpected result. Expected: %v, but got %v", expected, result)
+	}
+}
+
+func TestDotProduct(t *testing.T) {
+	v1 := []float32{1, 2, 3}
+	v2 := []float32{2, 4, 6}
+	expected := float32(28.0)
+	result := DotProduct(v1, v2)
+	if result != expected {
```

Comment:
You can't compare floats like that https://bitbashing.io/comparing-floats.html

Reply:
@sashabaranov  So how to compare? Can you provide some case parameters?

Reply:
@aceld something like

```go
func isClose(a, b float32) bool {
        if a == b {
		return true
	}
	return math.Abs(float64(a-b)) < 1e-12
}
```

https://floating-point-gui.de/errors/comparison/